### PR TITLE
SAK-43000 Give assessment settings expand/collapse all functions consistent icons and highlighting

### PR DESCRIPTION
**Pull request by brickatyourfeet**
https://jira.sakaiproject.org/browse/SAK-43000

Menus that are open will have arrows pointing down, and are highlighted. Menus that are closed will have arrows pointed to the right - unhighlighted. Expand and collapse all buttons will consistently open / close / highlight

### CODE REVIEW

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp [hunk 1]
```diff
@@ -212,17 +212,21 @@
...
+			$(“div#jqueryui-accordion > h3.ui-accordion-header > span”).removeClass(“ui-icon-triangle-1-e”).addClass(“ui-icon-triangle-1-s”);
+			$(“div#jqueryui-accordion > h3.ui-accordion-header”).addClass(“ui-accordion-header-active ui-state-active”);
 		}
 
 		function collapseAccordion(iframId){
 			$('.ui-accordion-content').hide();
 			mySetMainFrameHeight(iframId);
 			$("#collapseLink").hide();
 			$("#expandLink").show();
+			$(“div#jqueryui-accordion > h3.ui-accordion-header > span”).removeClass(“ui-icon-triangle-1-s”).addClass(“ui-icon-triangle-1-e”);
```
Comment (SedueRey) [inline samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp]: Hi, this quotes are not right, they are french quotes, you need to update them, thanks :)

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp [hunk 2]
```diff
@@ -212,17 +212,21 @@
         });
 
         function expandAccordion(iframId){
-			$('.ui-accordion-content').show();
+			$(‘div#jqueryui-accordion > .ui-accordion-content’).show();
 			mySetMainFrameHeight(iframId);
 			$("#collapseLink").show();
 			$("#expandLink").hide();
+			$(“div#jqueryui-accordion > h3.ui-accordion-header > span”).removeClass(“ui-icon-triangle-1-e”).addClass(“ui-icon-triangle-1-s”);
+			$(“div#jqueryui-accordion > h3.ui-accordion-header”).addClass(“ui-accordion-header-active ui-state-active”);
```
Comment (SedueRey) [inline samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp]: Same here

File: samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp [hunk 3]
```diff
@@ -182,10 +182,12 @@
           checkLastHandling();
         });
 		function expandAccordion(iframId){
-			$('.ui-accordion-content').show();
+			$(‘div#jqueryui-accordion > .ui-accordion-content’).show();
 			mySetMainFrameHeight(iframId);
 			$("#collapseLink").show();
 			$("#expandLink").hide();
+			$(“div#jqueryui-accordion > h3.ui-accordion-header > span”).removeClass(“ui-icon-triangle-1-e”).addClass(“ui-icon-triangle-1-s”);
+			$(“div#jqueryui-accordion > h3.ui-accordion-header”).addClass(“ui-accordion-header-active ui-state-active”);
```
Comment (SedueRey) [inline samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: same here, all quotes in this are french.

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp (hunk 1 above)
Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp]: There was both single quotes and double quotes in the functions already, so I just stuck with how they were. To be clear, which ones are French quotes? Should I be changing these to single or double quotes?

Comment (fostersdesign) [reply samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp]: I think @SedueRey was referring to the rounded-looking quotes `“` in comparison to the straight/vertical quotes `"` that were already there.

**Comment by brickatyourfeet** [comment]
> I wonder if there a simpler way to fix this so that the changes aren't so fragile and reliant on markup structure.
> 
> I would have to look into ideal ways to fix this, but I found this similar discussion:
> https://stackoverflow.com/questions/1592494/collapse-all-the-tabs-in-a-accordion-on-jquery
> that has a potentially-simple fix.

I found this same post when looking into this initially, and tried something of that nature, and had issues with it; I don't recall why though

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp (hunk 1 above)
Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp]: Oh, I see that now - will fix. Thanks.

**Comment by brickatyourfeet** [comment]
Fixed quotes, all straight now.

File: samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp [hunk 4]
```diff
@@ -182,10 +182,12 @@
           checkLastHandling();
         });
 		function expandAccordion(iframId){
-			$('.ui-accordion-content').show();
+			$("div#jqueryui-accordion > .ui-accordion-content").show();
 			mySetMainFrameHeight(iframId);
 			$("#collapseLink").show();
 			$("#expandLink").hide();
+			$("div#jqueryui-accordion > h3.ui-accordion-header > span").removeClass("ui-icon-triangle-1-e").addClass("ui-icon-triangle-1-s");
+			$("div#jqueryui-accordion > h3.ui-accordion-header").addClass("ui-accordion-header-active ui-state-active");
```
Comment (adrianfish) [inline samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: Why don't you change the icon in collapseAccordion here, too? Is there no need?

Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: It doesn't appear to be necessary

Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: This is how we've been running it in our instance, and it's done the trick

Comment (adrianfish) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: Ok, one more thing. I think you should move these functions into authoring.js. They are more or less identical. I know they weren't there in the first place, but it's a definite improvement in removing duplication. Never be afraid to improve code in the areas surrounding your bugfix.

Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: Not sure what you mean, are you saying get rid of them in publishedSettings and just leave them in authorSettings?

Comment (adrianfish) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: No, I mean move the function definition into authoring.js so the functions are defined in one place. Both JSPs bring that file in, so you can do that and still call it from the two jsps.

Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: Oh right right, thanks. They are now moved to authoring.js

Comment (adrianfish) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: Have you taken them out of the jsp files? It looks to me like they're still there. You can remove them from there and the call will use the definition from authoring.js.

**Comment by brickatyourfeet** [comment]
The jsp files are now amended with the functions in authoring.js

File: samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp (hunk 4 above)
Comment (brickatyourfeet) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp]: The jsp files are now amended with the functions in authoring.js

File: samigo/samigo-app/src/webapp/js/authoring.js [hunk 5]
```diff
@@ -787,4 +787,23 @@ function toggleCategories(checkbox) {
             categoryDiv.fadeOut();
         }
     }
-}
\ No newline at end of file
+}
+
+function expandAccordion(iframId){
+    $('div#jqueryui-accordion > .ui-accordion-content').show();
+    mySetMainFrameHeight(iframId);
```
Comment (fostersdesign) [inline samigo/samigo-app/src/webapp/js/authoring.js]: Is this frame height script still required now that we no longer have iFrames? Or does Lessons still require it for embedded quizzes?

File: samigo/samigo-app/src/webapp/js/authoring.js [hunk 6]
```diff
@@ -787,4 +787,23 @@ function toggleCategories(checkbox) {
...
+    $("#collapseLink").show();
+    $("#expandLink").hide();
+    $("div#jqueryui-accordion > h3.ui-accordion-header > span").removeClass("ui-icon-triangle-1-e").addClass("ui-icon-triangle-1-s");
```
Comment (fostersdesign) [inline samigo/samigo-app/src/webapp/js/authoring.js]: Sorry, I haven't investigated alternative ways of accomplishing this. The worry about these lines was how fragile they could be. We could file a separate Jira issue to fix them better in the future.

**Review by ern — APPROVED** [review]

**Pull request merged**